packages/model/lib/uriTemplateParser.js

```diff
@@ -40,6 +40,12 @@ class ParsedTemplateWrapper {
 
     return baseUri + this.toString()
   }
+
+  toRegex(baseUri) {
+    const absolute = this.toAbsoluteUrl(baseUri)
+
+    return '^' + absolute.replace(/{[^/]+}/g, '.+') + '$'
```

Comment:
Why is `[^/]+` replaced with `.+`?

Reply:
`/{[^/]+}/` matches template variables (excludes slash, otherwise it's greedy interpreting multiple variables as one)

Variable expressions are thus replaced with greedy patterns to match segments substituted with CSV values

Reply:
Ah, you made me realise that the pattern must also be `[^/]` otherwise it is also too greedy

Reply:
Oh I get it. And don't slashes cause issues if there are multiple variables replaced? I guess not since your tests are passing ;)

Reply:
Well, no, tests weren't good enough apparently :)

Reply:
The pattern was good in the end so I added only the missing tests to check